src/main/kotlin/org/rust/cargo/runconfig/command/RunClippyAction.kt

```diff
@@ -5,17 +5,38 @@
 
 package org.rust.cargo.runconfig.command
 
+import com.intellij.notification.NotificationType
 import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.progress.EmptyProgressIndicator
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.util.Computable
 import org.rust.cargo.icons.CargoIcons
+import org.rust.cargo.project.settings.toolchain
 import org.rust.cargo.runconfig.getAppropriateCargoProject
 import org.rust.cargo.toolchain.CargoCommandLine
 import org.rust.cargo.toolchain.run
+import org.rust.ide.actions.InstallComponentAction
+import org.rust.ide.notifications.showBalloon
 
 class RunClippyAction : RunCargoCommandActionBase(CargoIcons.CLIPPY) {
-
     override fun actionPerformed(e: AnActionEvent) {
         val project = e.project ?: return
         val cargoProject = getAppropriateCargoProject(e.dataContext) ?: return
+        val projectDirectory = cargoProject.manifest.parent ?: return
+
+        val isClippyInstalled = ProgressManager.getInstance().runProcess(Computable {
```

Comment:
But `ProgressManager#runProcess ` doesn't solve the initial problem. It runs a new process from the current thread, in our case in EDT. Let's better use `ProgressManager#runProcessWithProgressSynchronously`. Yes, it can show progress dialog, but as I understand it doesn't do it when execution is short